### PR TITLE
Refresh virtualization host pillar due to virtpoller removal

### PR DESCRIPTION
**Pull request by cbosdo**
## What does this PR change?

Refresh the virtualization host pillars after a migration from a server where `virtpoller` beacon was available.
Removes warning lines like the following from the salt minion logs of the virtual hosts:

```
"2021-07-13 23:13:37,359 [salt.beacons     :140 ][WARNING ][9588] Unable to process beacon virtpoller"
```

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [X] **DONE**

## Test coverage
- No tests: migration only, too much of a corner case

- [X] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/15428

- [X] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by hustodemon — APPROVED** [review]

File: java/code/src/com/redhat/rhn/manager/satellite/UpgradeCommand.java [hunk 1]
```diff
@@ -321,4 +331,23 @@ private void refreshCustomSlsFiles() {
...
+     */
+    private void refreshVirtHostPillar() {
+        try {
+            log.warn("Refreshing virtualization hosts pillar");
+            List<MinionServer> virtHosts = MinionServerFactory.listMinions().stream()
+                    .filter(Server::hasVirtualizationEntitlement)
+                    .collect(Collectors.toList());
+            virtHosts.forEach(MinionPillarManager.INSTANCE::generatePillar);
+            List<String> minionIds = virtHosts.stream().map(MinionServer::getMinionId).collect(Collectors.toList());
+            GlobalInstanceHolder.SALT_API.refreshPillar(new MinionList(minionIds));
```
Comment (cbosdo) [inline java/code/src/com/redhat/rhn/manager/satellite/UpgradeCommand.java:346]: This `refresh_pillar` is handled by the minion, but for some reason it doesn't get the updated data. Running `salt '*' saltutil.refresh_pillar` manually after the update works though. I need to investigate that before merging

Comment (cbosdo) [reply java/code/src/com/redhat/rhn/manager/satellite/UpgradeCommand.java:345]: The issue is a Salt one. Hacked around it by calling `saltutil.refresh_module` in `refreshPillar`
